README
NAME
    Task::WebGUI - Install WebGUI dependencies

VERSION
    version 1.101150

DESCRIPTION
    This Task bundles up all WebGUI prerequisites (currently targeting the
    WebGUI8 development branch).

    The preferred way of installing WebGUI is via the WRE (see
    <http://getwebgui.com>) however you may find this module useful in some
    circumstances (especially on dev servers).

    Even if you don't find it useful, we find the following CPAN Testers
    reports that are generated by the existence of this module useful!

    *   <http://www.cpantesters.org/distro/T/Task-WebGUI.html>

    *   <http://deps.cpantesters.org/?module=Task::WebGUI>

    Thanks CPAN Testers!

Image::Magick
    If Image::Magick fails to install, you may need to download it from
    <http://www.imagemagick.org> and install it manually:

     tar xvfz ImageMagick.tar.gz
     cd ImageMagick-*
     ./configure
     make && make check && sudo make install
     cd PerlMagick
     perl Makefile.PL
     make test && make install

SEE ALSO
    <http://webgui.org>

AUTHOR
      Patrick Donelan <pat@patspam.com>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Patrick Donelan.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Task-WebGUI

1.101150  2010-04-25 00:33:00 America/New_York
    Lots more dependencies

1.101110  2010-04-21 15:55:55 EST5EDT
    First version



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;

my %module_build_args = (
                       'dist_abstract' => 'Install WebGUI dependencies',
                       'script_files' => [],
                       'license' => 'perl',
                       'build_requires' => {
                                             'Test::More' => '0',
                                             'File::Find' => '0',
                                             'File::Temp' => '0',
                                             'Module::Build' => '0.3601'
                                           },
                       'dist_name' => 'Task-WebGUI',
                       'dist_author' => [
                                          'Patrick Donelan <pat@patspam.com>'
                                        ],
                       'module_name' => 'Task::WebGUI',
                       'requires' => {
                                       'HTML::Packer' => '0.4',
                                       'DateTime::Format::HTTP' => '0.38',
                                       'HTML::TagCloud' => '0.34',
                                       'HTML::Highlight' => '0.20',
                                       'XML::FeedPP' => '0.40',
                                       'MIME::Tools' => '5.427',
                                       'HTTP::Headers' => '1.61',
                                       'Text::CSV_XS' => '0.64',
                                       'File::Path' => '2.07',
                                       'Test::Deep' => '0.095',
                                       'List::MoreUtils' => '0.22',
                                       'DBI' => '1.607',
                                       'DateTime::Event::ICal' => '0.10',
                                       'Test::MockObject' => '1.02',
                                       'IO::Zlib' => '1.09',
                                       'Text::Aspell' => '0.01',
                                       'Net::POP3' => '2.29',
                                       'Memcached::libmemcached' => '0.3102',
                                       'Image::Magick' => '6.0',
                                       'Plack::Middleware::ForwardedHeaders' => '0',
                                       'DBD::mysql' => '4.010',
                                       'Path::Class' => '0.16',
                                       'Module::Find' => '0.06',
                                       'HTTP::Request' => '1.40',
                                       'Tie::CPHash' => '1.04',
                                       'POE' => '1.005',
                                       'Compress::Zlib' => '2.015',
                                       'URI::Escape' => '3.29',
                                       'HTML::Template' => '2.9',
                                       'MooseX::NonMoose' => '0.07',
                                       'Plack::Middleware::Status' => '0',
                                       'MooseX::Storage' => '0.23',
                                       'Color::Calc' => '0',
                                       'Net::Subnets' => '0.21',
                                       'CHI' => '0',
                                       'POE::Component::Client::HTTP' => '0.88',
                                       'Moose' => '0.93',
                                       'Plack::Middleware::Debug' => '0.07',
                                       'Test::Harness' => '3.17',
                                       'namespace::autoclean' => '0.09',
                                       'Plack' => '0.9929',
                                       'Locales' => '0.10',
                                       'Net::CIDR::Lite' => '0.20',
                                       'Archive::Any' => '0.0932',
                                       'Readonly' => '1.03',
                                       'Net::SMTP' => '2.31',
                                       'Config::JSON' => '1.5',
                                       'Params::Validate' => '0.91',
                                       'Net::LDAP' => '0.39',
                                       'DateTime' => '0.4501',
                                       'Archive::Zip' => '1.26',
                                       'DateTime::Format::Strptime' => '1.0800',
                                       'Finance::Quote' => '1.15',
                                       'List::Util' => '0',
                                       'Class::InsideOut' => '1.09',
                                       'Digest::SHA' => '5.47',
                                       'Crypt::SSLeay' => '0.57',
                                       'Archive::Tar' => '1.44',
                                       'JSON::Any' => '1.22',
                                       'Business::PayPal::API' => '0.62',
                                       'Pod::Coverage' => '0.19',
                                       'Cache::FastMmap' => '0',
                                       'Test::Exception' => '0.27',
                                       'Test::Class' => '0.31',
                                       'DateTime::Format::Mail' => '0.3001',
                                       'POE::Component::IKC::Server' => '0.2001',
                                       'Test::More' => '0.82',
                                       'HTML::Parser' => '3.60',
                                       'Weather::Com::Finder' => '0.5.3',
                                       'Log::Log4perl' => '1.20',
                                       'LWP' => '5.833',
                                       'Scope::Guard' => '0.03',
                                       'HTML::TagFilter' => '1.03',
                                       'Tie::IxHash' => '1.21',
                                       'XML::Simple' => '2.18',
                                       'JavaScript::Packer' => '0.04',
                                       'POSIX' => '0',
                                       'CSS::Packer' => '0.2',
                                       'Text::Balanced' => '2.00',
                                       'Digest::MD5' => '2.38',
                                       'CSS::Minifier::XS' => '0.03',
                                       'HTML::Template::Expr' => '0.07',
                                       'Class::C3' => '0.21',
                                       'Time::HiRes' => '1.9719',
                                       'Template' => '2.20',
                                       'Exception::Class' => '1.26',
                                       'Image::ExifTool' => '7.67',
                                       'Business::Tax::VAT::Validation' => '0.20',
                                       'JavaScript::Minifier::XS' => '0.05',
                                       'Clone' => '0.31',
                                       'JSON' => '2.12'
                                     },
                       'recommends' => {},
                       'configure_requires' => {
                                                 'ExtUtils::MakeMaker' => '6.31',
                                                 'Module::Build' => '0.3601'
                                               },
                       'dist_version' => '1.101150'
                     );


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Task/WebGUI.pm
# 
# This file is part of Task-WebGUI
# 
# This software is copyright (c) 2010 by Patrick Donelan.
# 
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
# 
package Task::WebGUI;
BEGIN {
  $Task::WebGUI::VERSION = '1.101150';
}

# ABSTRACT: Install WebGUI dependencies


1;

__END__
=pod

=head1 NAME

Task::WebGUI - Install WebGUI dependencies

=head1 VERSION

version 1.101150

=head1 DESCRIPTION

This Task bundles up all WebGUI prerequisites (currently targeting the WebGUI8
development branch).

The preferred way of installing WebGUI is via the WRE (see L<http://getwebgui.com>)
however you may find this module useful in some circumstances (especially
on dev servers).

Even if you don't find it useful, we find the following CPAN Testers
reports that are generated by the existence of this module useful!

=over 4

=item *

L<http://www.cpantesters.org/distro/T/Task-WebGUI.html>

=item *

L<http://deps.cpantesters.org/?module=Task::WebGUI>

=back

Thanks CPAN Testers!

=head1 Image::Magick

If Image::Magick fails to install, you may need to download it from 
L<http://www.imagemagick.org> and install it manually:

 tar xvfz ImageMagick.tar.gz
 cd ImageMagick-*
 ./configure
 make && make check && sudo make install
 cd PerlMagick
 perl Makefile.PL
 make test && make install

=head1 SEE ALSO

L<http://webgui.org>

=head1 AUTHOR

  Patrick Donelan <pat@patspam.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Patrick Donelan.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Task::WebGUI',
                       'DISTNAME' => 'Task-WebGUI',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.31',
                                                 'Module::Build' => '0.3601'
                                               },
                       'AUTHOR' => 'Patrick Donelan <pat@patspam.com>',
                       'BUILD_REQUIRES' => {
                                             'Test::More' => '0',
                                             'File::Find' => '0',
                                             'File::Temp' => '0',
                                             'Module::Build' => '0.3601'
                                           },
                       'ABSTRACT' => 'Install WebGUI dependencies',
                       'EXE_FILES' => [],
                       'VERSION' => '1.101150',
                       'PREREQ_PM' => {
                                        'HTML::Packer' => '0.4',
                                        'DateTime::Format::HTTP' => '0.38',
                                        'HTML::TagCloud' => '0.34',
                                        'HTML::Highlight' => '0.20',
                                        'XML::FeedPP' => '0.40',
                                        'MIME::Tools' => '5.427',
                                        'HTTP::Headers' => '1.61',
                                        'Text::CSV_XS' => '0.64',
                                        'File::Path' => '2.07',
                                        'Test::Deep' => '0.095',
                                        'List::MoreUtils' => '0.22',
                                        'DBI' => '1.607',
                                        'DateTime::Event::ICal' => '0.10',
                                        'Test::MockObject' => '1.02',
                                        'IO::Zlib' => '1.09',
                                        'Text::Aspell' => '0.01',
                                        'Net::POP3' => '2.29',
                                        'Memcached::libmemcached' => '0.3102',
                                        'Image::Magick' => '6.0',
                                        'Plack::Middleware::ForwardedHeaders' => '0',
                                        'DBD::mysql' => '4.010',
                                        'Path::Class' => '0.16',
                                        'Module::Find' => '0.06',
                                        'HTTP::Request' => '1.40',
                                        'Tie::CPHash' => '1.04',
                                        'POE' => '1.005',
                                        'Compress::Zlib' => '2.015',
                                        'URI::Escape' => '3.29',
                                        'HTML::Template' => '2.9',
                                        'MooseX::NonMoose' => '0.07',
                                        'Plack::Middleware::Status' => '0',
                                        'MooseX::Storage' => '0.23',
                                        'Color::Calc' => '0',
                                        'Net::Subnets' => '0.21',
                                        'CHI' => '0',
                                        'POE::Component::Client::HTTP' => '0.88',
                                        'Moose' => '0.93',
                                        'Plack::Middleware::Debug' => '0.07',
                                        'Test::Harness' => '3.17',
                                        'namespace::autoclean' => '0.09',
                                        'Plack' => '0.9929',
                                        'Locales' => '0.10',
                                        'Net::CIDR::Lite' => '0.20',
                                        'Archive::Any' => '0.0932',
                                        'Readonly' => '1.03',
                                        'Net::SMTP' => '2.31',
                                        'Config::JSON' => '1.5',
                                        'Params::Validate' => '0.91',
                                        'Net::LDAP' => '0.39',
                                        'DateTime' => '0.4501',
                                        'Archive::Zip' => '1.26',
                                        'DateTime::Format::Strptime' => '1.0800',
                                        'Finance::Quote' => '1.15',
                                        'List::Util' => '0',
                                        'Class::InsideOut' => '1.09',
                                        'Digest::SHA' => '5.47',
                                        'Crypt::SSLeay' => '0.57',
                                        'Archive::Tar' => '1.44',
                                        'JSON::Any' => '1.22',
                                        'Business::PayPal::API' => '0.62',
                                        'Pod::Coverage' => '0.19',
                                        'Cache::FastMmap' => '0',
                                        'Test::Exception' => '0.27',
                                        'Test::Class' => '0.31',
                                        'DateTime::Format::Mail' => '0.3001',
                                        'POE::Component::IKC::Server' => '0.2001',
                                        'Test::More' => '0.82',
                                        'HTML::Parser' => '3.60',
                                        'Weather::Com::Finder' => '0.5.3',
                                        'Log::Log4perl' => '1.20',
                                        'LWP' => '5.833',
                                        'Scope::Guard' => '0.03',
                                        'HTML::TagFilter' => '1.03',
                                        'Tie::IxHash' => '1.21',
                                        'XML::Simple' => '2.18',
                                        'JavaScript::Packer' => '0.04',
                                        'POSIX' => '0',
                                        'CSS::Packer' => '0.2',
                                        'Text::Balanced' => '2.00',
                                        'Digest::MD5' => '2.38',
                                        'CSS::Minifier::XS' => '0.03',
                                        'HTML::Template::Expr' => '0.07',
                                        'Class::C3' => '0.21',
                                        'Time::HiRes' => '1.9719',
                                        'Template' => '2.20',
                                        'Exception::Class' => '1.26',
                                        'Image::ExifTool' => '7.67',
                                        'Business::Tax::VAT::Validation' => '0.20',
                                        'JavaScript::Minifier::XS' => '0.05',
                                        'Clone' => '0.31',
                                        'JSON' => '2.12'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{BUILD_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.56) };

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



